pkg/cmd/infra/router/template.go

```diff
@@ -192,7 +195,11 @@ func (o *TemplateRouterOptions) Run() error {
 	}
 
 	statusPlugin := controller.NewStatusAdmitter(templatePlugin, oc, o.RouterName)
-	plugin := controller.NewUniqueHost(statusPlugin, o.RouteSelectionFunc(), statusPlugin)
+	var nextPlugin router.Plugin = statusPlugin
+	if o.ExtendedValidation {
+		nextPlugin = controller.NewExtendedValidator(nextPlugin, statusPlugin)
+	}
+	plugin := controller.NewUniqueHost(nextPlugin, o.RouteSelectionFunc(), statusPlugin)
```

Comment:
cast `statusPlugin` to RejectionRecorder here so we know we're not using its plugin functions